src/main/java/io/github/antishake/AntiShake.java

```diff
@@ -41,35 +39,37 @@
 
   // To load the config.properties when the class is loaded
   static {
-    InputStream is = null;
-    try {
-      properties = new Properties();
-      is = ClassLoader.class.getResourceAsStream("/config.properties");
-      properties.load(is);
-      loadProperties();
-    } catch (FileNotFoundException e) {
-      logger.error(e);
-    } catch (IOException e) {
-      logger.error(e);
-    }
+
   }
 
   public AntiShake(MotionCorrectionListener listener) {
+    this();
     this.motionCorrectionListener = listener;
 
     // As impulse response of Spring-Mass-Damper system is constant for the given SPRING_CONSTANT, we calculate only once while AntiShake object creation.
     calculateImplulseResponse();
   }
 
   // This constructor is not exposed with public access
-  // Used only for testing
   AntiShake() {
-
+    // TODO Optionally, get this resource from the constructor so that Android can provide it
+    InputStream is = null;
+    try {
+      properties = new Properties();
+      is = ClassLoader.class.getResourceAsStream("/config.properties");
+      properties.load(is);
+    } catch (FileNotFoundException | NullPointerException e) {
+//      logger.error(e);
+    } catch (IOException e) {
+//      logger.error(e);
+    }
+    loadProperties();
   }
 
   /**
    * Gets called by the client application with accelerometer values
    * to calculate transformation vector
+   * TODO Make this function use Coordinate instead
```

Comment:
Client applications may not have the Coordinate that library has. So, we can get the data in three aces and form a Coordinate at library end.